R/treeview.R

```diff
@@ -1,394 +1,463 @@
 		}
 	}
-	tr2 = keep.tip( tr1, na.omit( sc0$representative ) )
+	tr2 <- ape::keep.tip(tr1,
+	                     stats::na.omit(sc0$representative))
 
 	# subtrees
-	message( 'Computing sub-trees' )
-	stres2 <- subtrees( tr2, wait = TRUE )
-	Nstres2 <- sapply( stres2, Ntip )
+	message("Computing sub-trees")
+	stres2 <- ape::subtrees(tr2, wait = TRUE)
+	Nstres2 <- sapply(stres2, ape::Ntip)
 
 	# for each cluster find the node in tr2 representing it
 	## internal nodes first
-	for ( i in seq_along( stres2 ) ){
-		inode = i + Ntip( tr2 )
-		uv = na.omit( sc0$node_number[ match( stres2[[i]]$tip.label , sc0$representative ) ]  )
-		shared_anc = Reduce( intersect, e0$ancestors[uv] )
-		shared_anc2 = setdiff( intersect ( shared_anc , sc0$node_number ), uv )
-		if ( length( shared_anc2 ) > 0 ){
-			a = shared_anc2[ which.min( e0$ndesc[shared_anc2] ) ]
-			sc0$tr2mrca[ sc0$node_number == a ] <- inode
+	for (i in seq_along(stres2)){
+		inode <- i + ape::Ntip(tr2)
+		uv <- stats::na.omit(sc0$node_number[match(stres2[[i]]$tip.label,
+		                                    sc0$representative)])
+		shared_anc <- Reduce(intersect,
+		                     e0$ancestors[uv])
+		shared_anc2 <- setdiff(intersect(shared_anc,
+		                                 sc0$node_number),
+		                       uv)
+		if (length(shared_anc2) > 0) {
+			a <- shared_anc2[which.min(e0$ndesc[shared_anc2])]
+			sc0$tr2mrca[sc0$node_number == a] <- inode
 		}
 	}
 	## tips (takes precedence if overlap in tr2mrca )
-	i <- which( sc0$representative %in% tr2$tip.label )
-	sc0$tr2mrca[i] <- match( sc0$representative[i], tr2$tip.label )
+	i <- which(sc0$representative %in% tr2$tip.label)
+	sc0$tr2mrca[i] <- match(sc0$representative[i],
+	                        tr2$tip.label)
 
 	# tree data frames
 	## tips
-	sc2 <- sc0[ !is.na( sc0$tr2mrca ) , ]
-	sc2$date_range <- sapply( 1:nrow( sc2 ) , function(i) glue( '{sc2$least_recent_tip[i]} -> {sc2$most_recent_tip[i]}') )
-	tdvars = unique( c(branch_cols, 'logistic_growth_rate', 'clock_outlier', 'cluster_size', 'date_range', 'cluster_id', 'region_summary', 'cocirc_lineage_summary', 'lineage' , 'tr2mrca')  )
-	td0 <- sc2[ sc2$tr2mrca <= Ntip( tr2 )
-	 , tdvars
-	]
-	td0$lineages = td0$lineage
-	td0$cocirc_summary = td0$cocirc_lineage_summary
-	td0$node = td0$tr2mrca
-	td0$internal = 'N'
+	sc2 <- sc0[!is.na(sc0$tr2mrca), ]
+	sc2$date_range <- sapply(seq_len(nrow(sc2)),
+	                         function(i) glue::glue("{sc2$least_recent_tip[i]} -> {sc2$most_recent_tip[i]}"))
+	tdvars <- unique(c(branch_cols,
+	                 "logistic_growth_rate",
+	                 "clock_outlier",
+	                 "cluster_size",
+	                 "date_range",
+	                 "cluster_id",
+	                 "region_summary",
+	                 "cocirc_lineage_summary",
+	                 "lineage",
+	                 "tr2mrca"))
+	td0 <- sc2[sc2$tr2mrca <= ape::Ntip(tr2), tdvars]
```

Comment:
I'm trying not to comment too much on the state of the existing code, and just on any functional changes that have been introduced during the refactoring. Here, there's explicit duplication that could be simplified with an inner function.

Reply:
Next stage is refactoring the treeview() function, and splitting up into smaller functions...